src/server/scripts/EasternKingdoms/Stratholme/boss_baroness_anastari.cpp

```diff
@@ -88,37 +94,74 @@ class boss_baroness_anastari : public CreatureScript
             if (!UpdateVictim())
                 return;
 
-            //BansheeWail
-            if (BansheeWail_Timer <= diff)
-            {
-                if (rand32() % 100 < 95)
-                    DoCastVictim(SPELL_BANSHEEWAIL);
-                //4 seconds until we should cast this again
-                BansheeWail_Timer = 4000;
-            } else BansheeWail_Timer -= diff;
-
-            //BansheeCurse
-            if (BansheeCurse_Timer <= diff)
-            {
-                if (rand32() % 100 < 75)
-                    DoCastVictim(SPELL_BANSHEECURSE);
-                //18 seconds until we should cast this again
-                BansheeCurse_Timer = 18000;
-            } else BansheeCurse_Timer -= diff;
-
-            //Silence
-            if (Silence_Timer <= diff)
+            _events.Update(diff);
+
+            while (uint32 eventId = _events.ExecuteEvent())
             {
-                if (rand32() % 100 < 80)
-                    DoCastVictim(SPELL_SILENCE);
-                //13 seconds until we should cast this again
-                Silence_Timer = 13000;
-            } else Silence_Timer -= diff;
+                Unit* possessTarget = SelectTarget(SELECT_TARGET_RANDOM, 1, 0, true, false);    // Random target to be possessed
```

Comment:
it should be inside of EVENT_SPELL_POSSESS

Reply:
I presume that will be `case EVENT_SPELL_POSSESS:`
But only the line above, or more related code?

Reply:
Only the line above. Same for the comment below.

Reply:
Very well. I can move line 101 into `case EVENT_SPELL_POSSESS:`.
I still need to know where line 102 goes before I move it.

Reply:
```cpp
case EVENT_INVISIBLE:
```